aiida/repository/backend/disk_object_store.py

```diff
@@ -99,3 +99,19 @@ def delete_object(self, key: str):
         """
         super().delete_object(key)
         self.container.delete_objects([key])
+
+    def get_object_hash(self, key: str) -> str:
+        """Return the hash of a object identified by key.
+
+        .. note::
+            This hash value is dependant on the repository implementation
+            and so may change for different repositories.
+            By default the algorithm is SHA-256
+
+        :param key: fully qualified identifier for the object within the repository.
+        :raise FileNotFoundError: if the file does not exist.
+
+        """
+        if self.container.hash_type != 'sha256':
+            return super().get_object_hash(key)
```

Comment:
I was struggling for a bit to understand why this conditional was necessary because the container always uses the hash of the content as the key, so it shouldn't matter what hash type is used. However, then I realized that this would be a problem higher up that expects the same hash of a file regardless of what type of repository it is stored in. I think this is not directly obvious and think this should probably be written in the `Repository.get_hash` method docstring.

This may cause problems in the future when another repo is introduced that uses a different hashing type and all of a sudden caching and other things that rely on hashes may start failing, no?

Reply:
Yes, indeed might not be obvious that this function *must* return a `sha256` hash (as this is what AiiDA expects, and should not change between different backends). Would be good to clarify.

Reply:
Changed docstring to assert that SHA-256 hashing must be returned.

> This may cause problems in the future when another repo is introduced that uses a different hashing type

This would only now happen if the repo class specifically overrode `AbstractRepositoryBackend.get_object_hash` and did not "respect" the method's docstring, which would then be a bug in that classes implementation